src/sleep/sleep.c

```diff
@@ -62,15 +62,31 @@ static int write_hibernate_location_info(void) {
                                        "Invalid hibernate type: %s", type);
 
         /* Only available in 4.17+ */
-        if (access("/sys/power/resume_offset", W_OK) < 0) {
+        if (access("/sys/power/resume_offset", R_OK) < 0) {
                 if (errno == ENOENT) {
                         log_debug("Kernel too old, can't configure resume offset, ignoring.");
                         return 0;
                 }
 
-                return log_debug_errno(errno, "/sys/power/resume_offset not writeable: %m");
+                return log_debug_errno(errno, "/sys/power/resume_offset not readable: %m");
         }
 
+        /* use resume_offset set by user if available */
+        r = read_one_line_file("/sys/power/resume_offset", &resume_offset);
+        if (r < 0)
+                log_warning_errno(r, "Error reading from /sys/power/resume_offset: %m");
+
+        r = safe_atou64(resume_offset, &offset);
+        if (r < 0)
+                log_warning_errno(r, "Unable to verify resume_offset; ignoring");
+        else if (offset > 0) {
+                log_debug("Using /sys/power/resume_offset value: %s", resume_offset);
+                return 0;
```

Comment:
When reading something and converting it to a number, we always use the converted value in messages. This has the advantage that in case our conversion is somehow confused, we will report the value that will actually be used later. In other words, something like `log_debug("...: %d", offset)` would be better.